hack/ensure-acr-login.sh

```diff
@@ -27,4 +27,10 @@ if [[ "${REGISTRY:-}" =~ capzci\.azurecr\.io ]]; then
     : "${AZURE_SUBSCRIPTION_ID:?Environment variable empty or not defined.}"
     az account set -s "${AZURE_SUBSCRIPTION_ID}"
     az acr login --name capzci
+    # TODO(mainred): When using ACR, `az acr login` impacts the authentication of `docker buildx build --push` when the
+    # ACR, capz in our case, has anonymous pull enabled.
+    # Use `docker login` as a suggested workaround and remove this target when the issue is resolved.
+    # Issue link: https://github.com/Azure/acr/issues/582
+    # Failed building link: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cloud-provider-azure/974/pull-cloud-provider-azure-e2e-ccm-capz/1480459040440979456
+    docker login -u "${AZURE_CLIENT_ID}" -p "${AZURE_CLIENT_SECRET}" capzci.azurecr.io
```

Comment:
since we are using secret environment variables here, can we make sure to `set +o xtrace` at the top of the script so they don't get accidentally printed to stdout in prow?